### PR TITLE
The coordinator manages policy in cape

### PR DESCRIPTION
**Pull request by justin1121**
Here we create a new client that connects to the coordinator through
HTTP endpoints and a graphql endpoint. The client authenticates with the
coordinator using a token which can be retrieved via the coordinator
CLI. E.g. `cape tokens create`. Here's an example of how you would
connect to the coordinator to get a policy:

```
from cape_privacy import Client, parse_policy

c = Client("http://localhost:8080")

c.login("<token>")

p = c.query_project("my-project")

policy = parse_policy(p)

print(policy)
```

### CODE REVIEW

File: cape_privacy/coordinator/auth/api_token.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from cape_privacy.coordinator.utils import base64
+
+SECRET_BYTES = 16
+VERSION = b"\x01"
+
+
+class APIToken:
+    """Represents an API token used to authenticate with the coordinator.
+
+    The format is: <token_id>,<base64 string>
```
Comment (StarfallProjects) [inline cape_privacy/coordinator/auth/api_token.py:10]:
```suggestion
    The format is: <token_id>, <base64 string>
```

File: setup.py [hunk 2]
```diff
@@ -5,6 +5,8 @@
 with open("README.md", "r", encoding="utf8") as fh:
   long_description = fh.read()
 
+print(setuptools.find_packages())
```
Comment (bendecoste) [inline setup.py:8]: remove?

File: cape_privacy/coordinator/auth/api_token.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    The first byte of the decoded base64 string is the version and the rest
```
Comment (StarfallProjects) [inline cape_privacy/coordinator/auth/api_token.py:12]:
```suggestion
    The first byte of the decoded Base64 string is the version and the rest
```

File: cape_privacy/coordinator/client.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+from typing import Dict
+
+import requests
+
+from cape_privacy.coordinator.auth.api_token import APIToken
+from cape_privacy.coordinator.utils import base64
+
+
+class GraphQLError:
+    """Represents a graphql error that can be returned by a coordinator.
```
Comment (StarfallProjects) [inline cape_privacy/coordinator/client.py:11]:
```suggestion
    """Represents a GraphQL error that can be returned by a coordinator.
```

File: cape_privacy/coordinator/client.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            j = r.json()
+        except ValueError:
+            r.raise_for_status()
+
+        self.token = base64.from_string(j["token"])
+
+        return self.token
+
+    def me(self) -> str:
+        """Returns the id of the authenticated identity."""
```
Comment (StarfallProjects) [inline cape_privacy/coordinator/client.py:115]:
```suggestion
        """Returns the ID of the authenticated identity."""
```

File: cape_privacy/coordinator/client.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+    def query_policies(self) -> [Dict[Any, Any]]:
+        """Queries all of the policies for the authenticated identity."""
+
+        id = self.me()
+
+        return self.identity_policies(id)
+
+    def get_policy(self, label: str):
+        """Returns the current policy for a given project label"""
```
Comment (StarfallProjects) [inline cape_privacy/coordinator/client.py:137]:
```suggestion
        """Returns the current policy for a given project label."""
```

File: cape_privacy/policy/policy.py [hunk 7]
```diff
@@ -94,7 +97,7 @@ def apply_policy(policy: data.Policy, df, inplace=False):
     return result_df
 
 
-def parse_policy(p: str):
+def parse_policy(p: Union[str, Dict[Any, Any]]):
     """Parses a policy yaml file.
```
Comment (StarfallProjects) [inline cape_privacy/policy/policy.py:101]:
```suggestion
    """Parses a policy YAML file.
```

File: cape_privacy/policy/policy.py [hunk 8]
```diff
@@ -94,7 +97,7 @@ def apply_policy(policy: data.Policy, df, inplace=False):
...
 
     The passed in string can either be a path to a local file or
```
Comment (StarfallProjects) [inline cape_privacy/policy/policy.py:103]: Looks like this sentence is unfinished?

File: cape_privacy/policy/policy.py [hunk 9]
```diff
@@ -106,15 +109,17 @@ def parse_policy(p: str):
     Returns:
         The Policy object initialized by the yaml.
```
Comment (StarfallProjects) [inline cape_privacy/policy/policy.py:110]:
```suggestion
        The Policy object initialized by the YAML.
```

File: setup.py (hunk 2 above)
Comment (justin1121) [reply setup.py]: Forgot to push.

**Comment by justin1121** [comment]
Rename what `coordinator`?

File: cape_privacy/coordinator/auth/api_token.py (hunk 1 above)
Comment (justin1121) [reply cape_privacy/coordinator/auth/api_token.py:10]: Hmm that's literally the format of the token "<token_id>" followed by a comma "," and then the "base64 string". Can you think of a better way to represent this?

File: cape_privacy/policy/policy.py (hunk 8 above)
Comment (justin1121) [reply cape_privacy/policy/policy.py:103]: Its continued on the line below. This needs to be updated though.

**Comment by bendecoste** [comment]
> Rename what `coordinator`?

rename from coordinator to something else sorry .. out of scope 🙈

File: cape_privacy/coordinator/auth/api_token.py (hunk 1 above)
Comment (StarfallProjects) [reply cape_privacy/coordinator/auth/api_token.py]: Yeah I was just adding a space after the comma

**Pull request merged**